# Assignment 6
* Author: **vinay reddy vangala**
* NetID: **vrv22001**
* Date: **02/21/2023**

Here is a blank notebook for Assignment 6. Please fill in your name and NetID at the top. This assignment is designed to introduce you to how to use R with Google Colab. After finishing all your answers, please click "Runtime-> Run all" (or Ctrl+F9) to generate all the outputs and submit your ipynb file with all the outputs.

In [1]:
version # make sure it says language R at the bottom!

               _                                          
platform       x86_64-pc-linux-gnu                        
arch           x86_64                                     
os             linux-gnu                                  
system         x86_64, linux-gnu                          
status         Patched                                    
major          4                                          
minor          2.2                                        
year           2022                                       
month          11                                         
day            10                                         
svn rev        83330                                      
language       R                                          
version.string R version 4.2.2 Patched (2022-11-10 r83330)
nickname       Innocent and Trusting                      

# General Instructions

Please change the file name to “Assignment 6 Solution.ipynb.” The solution to each problem should be a separate section. 

Note that R code submitted should work independent of the data that
sits in the data structure. For example, suppose there was a vector
$Rvec$ with the values $(1,1,4)$ and the problem asks for you to
create R code to create a vector answer which doubles each element
of $Rvec$. 


*   The answer $answer=c(2,2,8)$ would be given no credit.

*  The answer $answer=2*Rvec$ would be an appropriate answer. 

For all problems, be sure to add comments that will enable the graders
to understand your process, as needed. If you have any questions,
please submit them via email to the instructor and/or the graders
prior to submitting your solution. 

# Problem 1 (50 Points) 

Let $X$ be a random variable that is normally distributed with mean of 10 and standard deviation of 4. Let $Y$ be a random variable that is normally distributed with mean of 8 and standard deviation of 2. $X$ and $Y$ are independent of each other. Let $Z$ be a random variable that is $2X - 3Y$. 

Please use 1000 as the replication number and set seed = 1000 for any parts that require simulation by using the command `set.seed(1000)`.  
 


a. What is the expected value of $X$?



In [80]:
set.seed(1000)
x <- rnorm(1000, mean = 10, sd = 4)
mean(x) # estimated expected value of X
var(x)

[1] 9.950096

[1] 15.42123

b. What is the expected value of $Y$? 

In [81]:
set.seed(1000)
y <- rnorm(1000, mean = 8, sd = 2)
mean(y) # estimated expected value of Y
var(y)

[1] 7.975048

[1] 3.855309

c. What is the expected value of $Z$?

In [92]:
set.seed(1000)

# Simulate Z 1000 times
z <- replicate(1000, 2*rnorm(1, 10, 4) - 3*rnorm(1, 8, 2))


# Calculate the mean and standard deviation of Z
mean_z <- mean(z)
mean_z
#var_z= 4* var(x) + 9* var(y)-12*cov(x,y)
#Since $X$ and $Y$ are independent, their covariance is 0. 
var_z= 4* var(x) +9* var(y)
var_z
#sd
sd_z=sd(z)
sd_z



[1] -3.863489

[1] 96.38272

[1] 10.02143

d. What is the variance of $Y + Z$?

In [102]:
#Var(Y+Z)=Var(Y)+Var(Z)+2Cov(Y,Z)
set.seed(1000)
z=rnorm(1000,mean_z,sd_z)
var_y_z = var(y) + var(z) + 2*cov(y,z)
var_y_z

[1] 139.2873

e. Use simulation to estimate the correlation of $Y$ and $Z$. What is your estimate of the correlation? 

In [107]:
set.seed(1000)
sim_y= replicate(1000,mean(rnorm(1000,8,2)))
sd_y1=sd(sim_y)

sim_z= replicate(1000,mean(rnorm(1000,mean_z,sd_z)))
sd_z1=sd(sim_z)

cor_y_z=cov(y,z)/(sd_y1*sd_z1)
cor_y_z




[1] 1000.017

f. Simulate $Y+Z$. What is the estimated variance of $Y+Z$ based on the simulation?

In [120]:
set.seed(1000)
mean_sim_y=mean(sim_y)
mean_sim_z=mean(sim_z)
sd_y1
sd_z1
var_sim_y= sd_y1^2
var_sim_z= sd_z1^2
var_y_z= var_sim_y + var_sim_z + 2*cov(sim_y,sim_z)
var_y_z


[1] 0.06192755

[1] 0.3119375

[1] 0.0999394

# Problem 2 (50 points) 

Suppose that household income in Stamford, CT is normally distributed with mean of \$96,800 and standard deviation of \$10,000. Please answer the following problems and use command set.seed(1001) for any parts that require simulation.

a. Suppose I take a random sample of 10 individuals from the area.  What is the distribution of the **sample average salary**?  Please indicate the parameters of the distribution as well as the distribution type. 

If we take a random sample of 10 individuals from the area, then the distribution of the sample average salary will be approximately normal, with a mean of the population mean and a standard error of the population standard deviation divided by the square root of the sample size.

The parameters of the distribution of the sample average salary are:

Mean: 96800 (same as the population mean)

Standard deviation: 10000 / sqrt(10) = 3162.28

The distribution type is normal.

b. Simulate 1000 samples of size 10.  What is the average value of the sample mean? What is the standard deviation of the sample mean? Are they close to the true distribution parameters? Explain.

In [44]:
set.seed(1001)
sample_means <- replicate(1000, mean(rnorm(10, mean = 96800, sd = 10000)))
mean(sample_means)
sd(sample_means)

[1] 96796.55

[1] 3072.815

Based on the simulation results, we got an average value of the sample mean of 96796.55, which is very close to the true population mean of 96800. The standard deviation of the sample mean is 3072.8, which is also close to the true value of 3162.28. These results indicate that the sample mean is an unbiased estimator of the population mean, and the standard error of the sample mean decreases as the sample size increases.

c. Simulate 100000 samples of size 10.  What is the average value of the sample mean? What is the standard deviation of the sample mean? Are they close to the true distribution parameters? How does that compare with your answer to the previous parts.

In [116]:
set.seed(10001)
sample_means <- replicate(100000, mean(rnorm(10, mean = 96800, sd = 10000)))
mean(sample_means)
sd(sample_means)

[1] 96797.36

[1] 3163.445

Compared to the previous parts, the estimates of the sample mean and standard deviation based on 100000 samples are more precise and accurate than those based on only 1000 samples. This is expected because the standard error of the sample mean decreases as the sample size increases, and the law of large numbers ensures that the sample mean approaches the population mean as the sample size becomes larger.

d. Now suppose you take a random sample of size 30. What are the mean and standard deviation of sample average salary? 

In [119]:
# Mean of the sample average salary: Same as the population mean, which is 96800
# Standard deviation of the sample average salary: 
# Standard deviation of the population divided by the square root of the sample size, which is $10,000 / sqrt(30) = $1825.742
set.seed(10001)
sd_samples=10000
sd_size_30=sd_samples/sqrt(30)
sd_size_30

res2=rnorm(30,96800,10000)
mean(res2)
sd(res2)

[1] 1825.742

[1] 99727.77

[1] 8337.652

e. Simulate 100000 samples of size 30. What is the mean and standard deviation of the sample average salary? How are the sample mean and standard deviation different from the sample size of 10?

In [13]:
set.seed(1001)
sample_means <- replicate(100000, mean(rnorm(30, mean = 96800, sd = 10000)))
mean(sample_means)
sd(sample_means)

[1] 96805.11

[1] 1830.299

Compared to the sample size of 10, the mean of the sample average salary remains the same because the population mean is unchanged. However, the standard deviation of the sample average salary is smaller than that of the sample size of 10, which is expected because the standard error of the sample mean decreases as the sample size increases.

Therefore, the sample mean and standard deviation are different from the sample size of 10 because the sample size affects the precision and accuracy of the sample mean estimator. As the sample size increases, the sample mean becomes a more reliable estimate of the population mean, and the standard error of the sample mean decreases, resulting in a smaller standard deviation.